api/ping.go

```
@@ -0,0 +1,81 @@
+	conn.SetReadDeadline(time.Now().Add(60 * time.Second))
+	conn.SetPongHandler(func(string) error {
+		conn.SetReadDeadline(time.Now().Add(60 * time.Second))
+		return nil
+	})
+
+	// Start ping-pong mechanism
+	go func() {
+		ticker := time.NewTicker(30 * time.Second)
+		defer ticker.Stop()
+
+		for {
```

golangci-lint (1.64.8)

S1000: should use for range instead of for { select {} }

(gosimple)